### PR TITLE
Feature/clean branch name

### PR DESCRIPTION
**Pull request by fceller**

### CODE REVIEW

File: jenkins/helper/prepareOskar.ps1 [hunk 1]
```diff
@@ -22,6 +22,19 @@ If(-Not(Test-Path -PathType Container -Path "$HDD\$env:NODE_NAME"))
 $OSKARDIR = "$HDD\$env:NODE_NAME"
 Set-Location $OSKARDIR
 
+If($env:OSKAR_BRANCH)
+{
+    $env:OSKAR_BRANCH = $env:OSKAR_BRANCH -replace '[^a-zA-Z0-9#/+_-]', ''
```
Comment (KVS85) [inline jenkins/helper/prepareOskar.ps1]: Need to add dot possibility also.

File: jenkins/helper/prepareOskar.ps1 [hunk 2]
```diff
@@ -22,6 +22,19 @@ If(-Not(Test-Path -PathType Container -Path "$HDD\$env:NODE_NAME"))
...
+}
+If($env:ARANGODB_BRANCH)
+{
+    $env:ARANGODB_BRANCH = $env:ARANGODB_BRANCH -replace '[^a-zA-Z0-9#/+_-]', ''
```
Comment (KVS85) [inline jenkins/helper/prepareOskar.ps1]: Need to add dot possibility also.

File: jenkins/helper/prepareOskar.ps1 [hunk 3]
```diff
@@ -22,6 +22,19 @@ If(-Not(Test-Path -PathType Container -Path "$HDD\$env:NODE_NAME"))
...
+}
+If($env:ENTERPRISE_BRANCH)
+{
+    $env:ENTERPRISE_BRANCH = $env:ENTERPRISE_BRANCH -replace '[^a-zA-Z0-9#/+_-]', ''
```
Comment (KVS85) [inline jenkins/helper/prepareOskar.ps1]: Need to add dot possibility also.

File: jenkins/helper/jenkins.fish [hunk 4]
```diff
@@ -34,3 +34,22 @@ function prepareOskar
     end
   end
 end
+
+function cleanBranchName
+  echo $argv[1] | sed -e 's:[^-a-zA-Z0-9_/#.+]::g'
```
Comment (KVS85) [inline jenkins/helper/jenkins.fish]: Shoudn't dot be escaped here?

**Comment by KVS85** [comment]
@fceller FYI (thanks to @goedderz): https://git-scm.com/docs/git-check-ref-format

File: jenkins/helper/prepareOskar.ps1 (hunk 1 above)
Comment (fceller) [reply jenkins/helper/prepareOskar.ps1]: Done

File: jenkins/helper/prepareOskar.ps1 (hunk 2 above)
Comment (fceller) [reply jenkins/helper/prepareOskar.ps1]: Done

File: jenkins/helper/jenkins.fish (hunk 4 above)
Comment (fceller) [reply jenkins/helper/jenkins.fish]: Both work:

```
fceller@c9:~/oskar$ echo "Hal.lo" | sed -e 's:[^-a-zA-Z0-9_/#.+]::g'
Hal.lo
fceller@c9:~/oskar$ echo "Hal.lo" | sed -e 's:[^-a-zA-Z0-9_/#\.+]::g'
Hal.lo

```

File: jenkins/helper/prepareOskar.ps1 (hunk 3 above)
Comment (fceller) [reply jenkins/helper/prepareOskar.ps1]: done

**Comment by fceller** [comment]
We want stricter rules (eg. no UTF8) because this breaks Jenkins.

**Pull request merged**